File: kmp/core/domain/src/commonMain/kotlin/dev/mjablonski/rewordme/domain/SelectionFilter.kt
package dev.mjablonski.rewordme.domain

/**
 * Decides whether a selection is worth acting on: rejects empty strings,
 * stray whitespace, punctuation runs, lone symbols, bare numbers.
 */
object SelectionFilter {
    fun isMeaningful(text: String): Boolean {
        val trimmed = text.trim()
        if (trimmed.length < 3) return false
        return trimmed.any(Char::isLetter)
    }
}

File: Sources/RewordMeDomain/SelectionFilter.swift
import Foundation

/// Decides whether a selection is worth popping up for. Filters out the
/// noise a mouse produces all day: empty strings, stray whitespace,
/// punctuation runs, lone symbols, bare numbers.
public enum SelectionFilter {
    public static func isMeaningful(_ text: String) -> Bool {
        let trimmed = text.trimmingCharacters(in: .whitespacesAndNewlines)
        guard trimmed.count >= 3 else { return false }
        guard trimmed.rangeOfCharacter(from: .letters) != nil else { return false }
        return true
    }
}
